### PR TITLE
Authentication support for mlflow sync

### PR DESCRIPTION
**Pull request by gmertes**
Adds the `--authentication (-a)` option to the `mlflow sync` subcommand. This enables authentication on the destination server. It retrieves a token and sets it for the mlflow API to use during sync.

Tested the following: local logs -> new server, and old server -> new server (the latter only works from Atos).

Todo: test from Leonardo.

Usage:

`anemoi-training mlflow sync -s /path/to/logs -d https://mlflow.server -r 1234 -e sync-test -a`

### CODE REVIEW

**Review by JesperDramsch — DISMISSED** [review]
LGTM, thanks for implementing this!

**Review by anaprietonem — APPROVED** [review]
Approving to unblock Jesper !

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Keep it human-readable, your future self will thank you!
 ### Changed
 
 - Updated configuration examples in documentation and corrected links - [#46](https://github.com/ecmwf/anemoi-training/pull/46)
+- Feature: Authentication support for mlflow sync
```
[the remark this answers is not in the document]
Comment (gmertes) [reply CHANGELOG.md]: done

File: src/anemoi/training/diagnostics/mlflow/__init__.py [hunk 2]
```diff
@@ -4,3 +4,32 @@
 # In applying this licence, ECMWF does not waive the privileges and immunities
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
+
+import os
```
Comment (anaprietonem) [inline src/anemoi/training/diagnostics/mlflow/__init__.py:8]: are we happing putting this in the __init__ or should it go into a utils.py script for more general functions related to mlflow? No strong opinions in any case!

**Review by anaprietonem — DISMISSED** [review]
This PR LGTM! We have tested it both from ATOS and Leonardo and the command works and nicely catches if someone misses the authentication flag. Thanks for implementing this Gert!

File: src/anemoi/training/diagnostics/mlflow/__init__.py (hunk 2 above)
Comment (gmertes) [reply src/anemoi/training/diagnostics/mlflow/__init__.py]: I added a utils.py 😃

**Pull request merged**